### PR TITLE
[Refactor] Remove limitation of groovy script

### PR DESCRIPTION
**Pull request by gengjun-git**
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/qe/ExecuteScriptExecutor.java [hunk 1]
```diff
@@ -51,9 +51,6 @@ static ShowResultSet makeResultSet(String result) {
     }
 
     public static ShowResultSet execute(ExecuteScriptStmt stmt, ConnectContext ctx) throws UserException {
-        if (!"127.0.0.1".equals(ctx.getRemoteIP()) || Config.enable_remote_script) {
-            throw new UserException("script can only be executed on localhost");
-        }
         if (stmt.isFrontendScript()) {
             return executeFrontendScript(stmt, ctx);
         } else {
```
Comment (decster) [inline fe/fe-core/src/main/java/com/starrocks/qe/ExecuteScriptExecutor.java:56]: just only remove limit for BE script? not FE(groovy script)?

Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/qe/ExecuteScriptExecutor.java:56]: There is already a config param `enable_execute_script_on_frontend` for fe.

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by decster — APPROVED** [review]

**Review by trueeyu — APPROVED** [review]

**Comment by gengjun-git** [comment]
@mergifyio backport branch-2.5 branch-3.0 branch-3.1 branch-3.2

**Pull request merged**